checkov/terraform/deep_analysis_plan_graph_manager.py

```diff
@@ -5,45 +5,70 @@
         self._apply_address_mapping()
 
     def _apply_address_mapping(self) -> None:
-        self._address_to_tf_vertex_map = {
-            vertex.attributes[TF_PLAN_RESOURCE_ADDRESS]: vertex
-            for vertex in self.tf_graph.vertices
+        self._address_to_tf_idx_and_vertex_map = {
+            vertex.attributes[TF_PLAN_RESOURCE_ADDRESS]: (i, vertex)
+            for i, vertex in enumerate(self.tf_graph.vertices)
             if vertex.block_type == BlockType.RESOURCE
         }
-        self._address_to_tf_plan_vertex_map = {
-            vertex.attributes[TF_PLAN_RESOURCE_ADDRESS]: vertex
-            for vertex in self.tf_plan_graph.vertices
+        self._address_to_tf_plan_idx_and_vertex_map = {
+            vertex.attributes[TF_PLAN_RESOURCE_ADDRESS]: (i, vertex)
+            for i, vertex in enumerate(self.tf_plan_graph.vertices)
             if vertex.block_type == BlockType.RESOURCE
         }
 
+    def append_vertex_to_terraform_graph(self, tf_plan_vertex, tf_plan_vertex_index, address):
+        new_vertex_idx = len(self.tf_graph.vertices)
+        self.tf_graph.vertices.append(tf_plan_vertex)
+        self._address_to_tf_idx_and_vertex_map[address] = (new_vertex_idx, tf_plan_vertex)
+
+        def __get_tf_vertex_idx_from_tf_plan_vertex(v):
+            v = self._address_to_tf_idx_and_vertex_map.get(v.attributes.get('__address__'))
+            if not v:
+                return None
+            return v[1]
+
+        for edge in self.tf_plan_graph.out_edges[tf_plan_vertex_index]:
+            dest = self.tf_plan_graph.vertices[edge.dest]
+            dest_index = __get_tf_vertex_idx_from_tf_plan_vertex(dest)
+            if dest_index:
+                self.tf_graph.create_edge(new_vertex_idx, dest_index, edge.label)
+        for edge in self.tf_plan_graph.in_edges[tf_plan_vertex_index]:
+            origin = self.tf_plan_graph.vertices[edge.origin]
+            origin_index = __get_tf_vertex_idx_from_tf_plan_vertex(origin)
+            if origin_index:
+                self.tf_graph.create_edge(origin_index, new_vertex_idx, edge.label)
+
     def enrich_tf_graph_attributes(self) -> None:
-        for address, tf_plan_vertex in self._address_to_tf_plan_vertex_map.items():
-            tf_vertex = self._address_to_tf_vertex_map.get(address)
-            if not tf_vertex:
-                logging.info(f'Cant find this address: {address} in tf graph')
+        for address, tf_plan_idx_and_vertex in self._address_to_tf_plan_idx_and_vertex_map.items():
+            tf_plan_vertex_index, tf_plan_vertex = tf_plan_idx_and_vertex
+            tf_idx_and_vertex = self._address_to_tf_idx_and_vertex_map.get(address)
+            if not tf_idx_and_vertex:
+                logging.info(f'Cant find this address: {address} in tf graph, adding it')
+                self.append_vertex_to_terraform_graph(tf_plan_vertex, tf_plan_vertex_index, address)
```

Comment:
aren't you adding vertices to the `terraform_plan` graph instead of `terraform`?
If you are doing it like you wrote, can you explain me why?

Reply:
The terraform graph contains more data on the variables, locals, edges and etc so it's better to run the queries on it. We are adding the missing resources from the terraform_plan graph for the edge case that the terraform graph won't contain it